### PR TITLE
pick the latest chain config along with block-level rules

### PR DESCRIPTION
**Pull request by sudeepdino008**
- pick the latest ChainConfig
- get the "block rules" based on passed block number and timestamp

### CODE REVIEW

**Review by rogarcia — APPROVED** [review]
at high level LGTM (also had this running on my operator since Friday without issues)
Let's get another review from @noslav before merging

**Review by noslav — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**
